src/utils/validations/validator/validator.js

```diff
@@ -0,0 +1,21 @@
+const isLegacy = (func) => { return typeof func === 'object'; };
+
+const handleLegacyValidation = (func, value, props) => {
+  if (func.validate(value, props)) {
+    return Promise.resolve();
+  }
+  return Promise.reject(func.message());
+};
+
+const validator = validationFunctions => (value, props) => {
+  const validations = Array.isArray(validationFunctions) ? validationFunctions : [validationFunctions];
+  const results = validations.map((func) => {
+    if (isLegacy(func)) {
+      return handleLegacyValidation(func, value, props);
+    }
+    return func(value, props);
+  });
```

Comment:
What do you think about something like:
```
const handleLegacyValidation = (func, value, props) => func.validate(value, props) ? Promise.resolve() : Promise.reject(func.message());
const handleValidation = (func) => isLegacy(func) ? handleLegacyValidation(func, value, props) : func(value, props);
...
const results = validations.map(handleValidation);
```

Reply:
I like the suggestion for `handleValidation` function so I've added it, I've not changed the `handleLegacyValidation` to ternary one as I feel it's too complex and makes it less readable